### ISSUE TITLE
No checks in place for broken hierarchy

### ISSUE THREAD

**Issue by krzysiula3000** [Sollumz/Sollumz#958]
a hierarchy consisting of a drawable and bound poly mesh gives an "XXX has no sollumz materials... skipping" instead of "hierarchy incorrect" or something

same thing for other combinations, but eg. clip dictionary and clip (no "clips" in between them) gives an exception error

so basically you need to write code checking every combination 😊😊👍

**Comment by DarkWolf197** [comment]
i don't think it needs to check for every possible combination. maybe just for one single expected hierarchy. for every main type 
```
    parent_types = [SollumType.FRAGMENT, SollumType.DRAWABLE, SollumType.DRAWABLE_DICTIONARY,
                    SollumType.CLIP_DICTIONARY, SollumType.YMAP, SollumType.BOUND_COMPOSITE]
```
make an _expected hierarchy_ that checks if maybe a drawable has only drawable models and lights as children, same for clip dictionaries, check if there are only CLIPS and ANIMATIONS as first-level child
and then check if there are only CLIP and ANIMATION objs in the CLIPS and ANIMATIONS children
